gameobjects/opponent.h

```diff
@@ -10,10 +11,17 @@ namespace pong
 
 class Ball;
 
-class Opponent : public GameObject
+class OpponentBlueprint : public GameObjectBlueprint
 {
 public:
+    OpponentBlueprint();
     void InitalizeComponents() override;
+};
+
+class Opponent : public Behavior
```

Comment:
This seems like possibly a bad inheritance hierarchy. Does it make sense that an Opponent "is-a" behavior?

it seems like an opponent may "have-a" behavior.

Reply:
Yeah I understand that. It may just be a game engine naming thing. In Unity the script components are called "MonoBehavior." Generally I have always named things in similar style. The entire player entity/object may have colliders, meshes, lights, multiple monobehaviors, but then also has the Player MonoBehavior to control movement and other things exclusive to the player. It could be renamed to something like PlayerScript or PlayerBehavior, but I think it generally gets the idea across that it is the main controller of the object's behavior.